src/software/ai/hl/stp/play/ball_placement_play_test.py

```diff
@@ -81,22 +86,174 @@ def test_two_ai_ball_placement(simulated_test_runner, run_enemy_ai):
+    # ]
+
+    # simulated_test_runner.run_test(
+    #     eventually_validation_sequence_set=drop_ball_eventually_validation_sequence_set,
+    #     always_validation_sequence_set=drop_ball_always_validation_sequence_set,
+    #     test_timeout_s=TEST_DURATION,
+    # )
+
+
+@pytest.mark.parametrize("run_enemy_ai", [False, True])
+def test_force_start_ball_placement(simulated_test_runner, run_enemy_ai):
+
+    # starting point must be Point
+    ball_initial_pos = tbots.Point(2, 2)
+    # placement point must be Vector2 to work with game controller
+    ball_final_pos = tbots.Point(-3, -2)
+
+    # Setup Bots
+    blue_bots = [
+        tbots.Point(-2.75, 2.5),
+        tbots.Point(-2.75, 1.5),
+        tbots.Point(-2.75, 0.5),
+        tbots.Point(-2.75, -0.5),
+        tbots.Point(-2.75, -1.5),
+        tbots.Point(4.6, -3.1),
+    ]
+    yellow_bots = [
+        tbots.Point(1, 0),
+        tbots.Point(1, 2.5),
+        tbots.Point(1, -2.5),
+        tbots.Field.createSSLDivisionBField().enemyGoalCenter(),
+        tbots.Field.createSSLDivisionBField().enemyDefenseArea().negXNegYCorner(),
+        tbots.Field.createSSLDivisionBField().enemyDefenseArea().negXPosYCorner(),
+    ]
+
+    # Game Controller Setup
+    simulated_test_runner.gamecontroller.send_ci_input(
+        gc_command=Command.Type.STOP, team=Team.UNKNOWN
+    )
+    simulated_test_runner.gamecontroller.send_ci_input(
+        gc_command=Command.Type.FORCE_START, team=Team.BLUE
     )
+    # Pass in placement point here - not required for all play tests
+    simulated_test_runner.gamecontroller.send_ci_input(
+        gc_command=Command.Type.BALL_PLACEMENT,
+        team=Team.BLUE,
+        final_ball_placement_point=ball_final_pos,
+    )
+
+    # Force play override here
+    blue_play = Play()
+    blue_play.name = PlayName.BallPlacementPlay
+
+    simulated_test_runner.blue_full_system_proto_unix_io.send_proto(Play, blue_play)
+
+    # We can parametrize running in ai_vs_ai mode
+    if run_enemy_ai:
+        yellow_play = Play()
+        yellow_play.name = PlayName.EnemyBallPlacementPlay
+
+        simulated_test_runner.yellow_full_system_proto_unix_io.send_proto(
+            Play, yellow_play
+        )
+
+    # Create world state
+    simulated_test_runner.simulator_proto_unix_io.send_proto(
+        WorldState,
+        create_world_state(
+            yellow_robot_locations=yellow_bots,
+            blue_robot_locations=blue_bots,
+            ball_location=ball_initial_pos,
+            ball_velocity=tbots.Vector(0, 0),
+        ),
+    )
+
+    # Placement Always Validation
+    placement_always_validation_sequence_set = [[]]
+
+    # Placement Eventually Validation
+    placement_eventually_validation_sequence_set = [
+        [
+            # Ball should arrive within 5cm of placement point
+            BallEventuallyEntersRegion(regions=[tbots.Circle(ball_final_pos, 0.1)]),
+            RobotEventuallyEntersRegion(regions=[tbots.Circle(ball_final_pos, 0.1)]),
```

Comment:
The circle for the robot should be larger than the circle for the ball. Same thing for the other tests (including the commented out ones). 
Also, shouldn't the radius be 0.05 instead of 0.1 (based on the comment and the SSL rules)?

Reply:
According to the ssl rules, it seems like the radius is 0.15, should I change it to 0.15?

Reply:
Yeah I think we should do whatever the rule is. We should probably add the same comment from the above test that links to the rule, here as well. 
Also, on second thought, I'm not sure if `RobotEventuallyEntersRegion` is even needed for these tests. We might be overconstraining something that's not in the rules. The rule states
> there is no robot within 0.05 meters distance to the ball if the next command is a [free kick](https://robocup-ssl.github.io/ssl-rules/sslrules.html#_free_kick) for the placing team

So we probably want to just have the `RobotEventuallyExitsRegion` with 0.05 meters for after the ball placement is done (basically what's in the commented code)

Reply:
The first part of ball_placement (uncommented part) only make sure that the robot and the ball sends the ball into the placement region (ball_final_pos - 0.15, ball_final_pos + 0.15). The ball may not stop because the robot may not have dropped the ball yet. Verification of robot entering the region ensures that the ball is carried by the robot into the region. 

The second part of ball_placement includes the robot dropping the ball and leaving the region according to the rules (0.05 for free kick and 0.5 for other Referee Commands). The always verification ensures the ball stays in the ball_placement region. The eventually verification ensures the ball was dropped (so it stops) and the robot exits the region with the errors of placement accounted. 

I updated the code to reflect this and will push that once fix formatting script is done on my end.